src/main/host/process.c

```diff
@@ -690,6 +705,8 @@ void process_stop(Process* proc) {
 
     info("terminating process '%s'", process_getName(proc));
 
+    proc->killedByShadow = true;
```

Comment:
Are we sure this is only run when shadow kills the process, and not if the process exits unexpectedly and shadow tries to stop the process again later (at the stop time). The below `_process_terminate_threads()` has a `process_isRunning()` check before sending the `SIGKILL`, and I think I remember seeing the "terminating process" info log message twice for some processes. I'll run some tests to see if this is working as expected.

Reply:
So `process_stop()` is run and `proc->killedByShadow` is set even for processes that unexpectedly died (like the 'kill' process in the below test). I think the `proc->killedByShadow = true` should be moved somewhere else, maybe directly before the `kill()` call.

Reply:
Oof, good catch. Done!